packages/core/src/icon/utils/icon.svg-helpers.ts

```diff
@@ -4,52 +4,38 @@
 import { IconShapeCollection } from '../interfaces/icon.interfaces.js';
-import { getIconSvgClasses, getShapeClassname, IconDecorationClassnames } from './icon.classnames.js';
-
-export function getBadgeSvg(shapeClassname: string) {
-  return [
-    '<circle cx="30" cy="6" r="5"  class="',
-    [shapeClassname, IconDecorationClassnames.Badge].join(' '),
-    '" />',
-  ].join('');
-}
 
-export function getAlertSvg(shapeClassname: string) {
-  return [
-    '<path d="M26.85,1.14,21.13,11A1.28,1.28,0,0,0,22.23,13H33.68A1.28,1.28,0,0,0,34.78,11L29.06,1.14A1.28,1.28,0,0,0,26.85,1.14Z" class="',
-    [shapeClassname, IconDecorationClassnames.Alert].join(' '),
-    '"/>',
-  ].join('');
+export function hasAlertBadge(icon: CdsIcon) {
+  return icon.badge && (icon.badge === 'inherit-triangle' || icon.badge === 'warning-triangle');
 }
 
-export function decorateSvgWithClassnames(shapeName: string, shapeSvg: string) {
-  const shapeClassname = getShapeClassname(shapeName);
-  let transformedSvg = shapeSvg.split('/>').join(` class="${shapeClassname}"/>`);
-
-  switch (shapeName) {
-    case 'solidBadged':
-    case 'outlineBadged':
-      transformedSvg = transformedSvg.concat(getBadgeSvg(shapeClassname));
-      break;
-    case 'solidAlerted':
-    case 'outlineAlerted':
-      transformedSvg = transformedSvg.concat(getAlertSvg(shapeClassname));
-      break;
-    default:
-      break;
+export function getIconBadgeSVG(icon: CdsIcon) {
+  let badge = '';
+
+  if (icon.badge && hasAlertBadge(icon)) {
+    badge =
+      '<path d="M26.85 1.14L21.13 11a1.28 1.28 0 001.1 2h11.45a1.28 1.28 0 001.1-2l-5.72-9.86a1.28 1.28 0 00-2.21 0z" class="alert" />';
+  } else if (icon.badge) {
+    badge = '<circle cx="30" cy="6" r="5" class="badge" />';
   }
-  return transformedSvg;
+
+  return badge;
 }
 
-export function getIconSvgOpeningTag(icon: IconShapeCollection) {
-  const iconSvgViewboxSize = 36;
-  const iconSvgClasses = getIconSvgClasses(icon);
+export function getIconSVG(icon: CdsIcon) {
+  const iconShape = (ClarityIcons.registry[icon.shape] ?? ClarityIcons.registry['unknown']) as IconShapeCollection;
+  let shape = icon.solid && iconShape.solid ? iconShape.solid : iconShape.outline;
 
-  return `<svg version="1.1" class="${iconSvgClasses}" viewBox="0 0 ${iconSvgViewboxSize} ${iconSvgViewboxSize}" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" focusable="false" role="img" aria-label="">`;
-}
+  if (icon.badge && !hasAlertBadge(icon)) {
+    shape = icon.solid ? iconShape.solidBadged ?? shape : iconShape.outlineBadged ?? shape;
+  }
+
+  if (hasAlertBadge(icon)) {
+    shape = icon.solid ? iconShape.solidAlerted ?? shape : iconShape.outlineAlerted ?? shape;
+  }
 
-export function getIconSvgClosingTag(icon: IconShapeCollection): string {
-  const makeTypescriptHappy = icon;
-  return makeTypescriptHappy ? '</svg>' : '</svg>';
+  return shape;
```

Comment:
This function is the main change for the large performance gain, by using this and the `svg` template tag from lit we can conditionally render the svg which can reduce the number of svg nodes in the DOM by 5x